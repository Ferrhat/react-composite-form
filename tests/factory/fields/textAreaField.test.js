import React from 'react';
import {shallow, mount} from 'enzyme';
import TextAreaField from "../../../lib/factory/fields/TextAreaField";


describe('TextAreaField', () => {
    const mockValueUpdateHandler = jest.fn();
    const mockChangeHandler = jest.fn();
    const textAreaField = shallow(<TextAreaField valueUpdateHandler={mockValueUpdateHandler} changeHandler={mockChangeHandler} name={'testName'} />);

    it('renders properly', () => {
        expect(textAreaField).toMatchSnapshot();
    });

    it('contains a connected FormControl component', () => {
        expect(textAreaField.find('FormControl').length).toEqual(1);
    });

    it('should trigger a change event', () => {
        const event = {
            preventDefault() {},
            target: { name: 'testName', value: 'testValue' }
        };
        textAreaField.find('FormControl').simulate('change', event);
        expect(mockValueUpdateHandler).toBeCalledWith('testName', 'testValue');
        expect(mockChangeHandler).toBeCalledWith(event);
    });

});
