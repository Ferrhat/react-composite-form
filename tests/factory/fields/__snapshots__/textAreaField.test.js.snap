// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaField renders properly 1`] = `
<div>
  <FormControl
    bsClass="form-control"
    componentClass="textarea"
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="validationErrorMessage"
  />
</div>
`;
